apply wna16marlin kernel in moe weight only quantization

AniZpZ opened the pull request:
<!-- Thank you for your contribution! We appreciate it. The following guidelines will help improve your pull request and facilitate feedback. If anything is unclear, don't hesitate to submit your pull request and ask the maintainers for assistance. -->

## Motivation

apply wna16marlin kernel to speed up weight only quantization dsv3 model
we oberserve a decoding speed of over 100tps when bs=1 on 8*H20 platform
<img width="987" alt="image" src="https://github.com/user-attachments/assets/b5f86f8a-62c7-4729-b0e6-b4e5e6adc040" />

performance(based on 0.4.6)

| Concurrency | DS R1 FP8 Throughput | DS R1 FP8 TTFT | DS R1  FP8 ITL | DS R1  FP8 E2E | This PR Throughput | This PR TTFT | This PR ITL | This PR E2E | Lift   |
|-------------|----------------|----------|---------|---------|-----------------|-----------|----------|----------|--------|
| 1           | 66.36          | 151.63   | 14.94   | 15430.27| 108.09          | 160.30    | 9.11     | 9472.19  | 62.88% |
| 2           | 112.02         | 239.62   | 17.64   | 18279.83| 150.22          | 238.39    | 13.10    | 13631.87 | 34.10% |
| 4           | 196.29         | 186.09   | 20.22   | 20862.50| 249.53          | 98.81     | 15.95    | 16409.83 | 27.12% |
| 8           | 329.92         | 340.68   | 23.94   | 24819.04| 386.28          | 112.75    | 20.62    | 21192.90 | 17.08% |
| 16          | 507.90         | 431.63   | 30.13   | 31238.33| 554.05          | 130.42    | 27.93    | 28691.76 | 9.08%  |
| 32          | 742.78         | 566.84   | 39.50   | 40956.04| 762.38          | 153.72    | 39.22    | 40260.41 | 2.64%  |


## Modifications

update route of compressed_tensors_moe and fix some bugs
Co-author: @huangtingwei9988 

Note: [PR553](https://github.com/sgl-project/sglang/pull/5535) by @yych0745 @HandH1998 @sleepcoo  add wna16 marlin moe to sglang which removes the dependency on vllm

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

yych0745 committed:
moe w4a16 marlin kernel cmake build

Co-authored-by: sleepcoo <sleepcoo@gmail.com>
Co-authored-by: HandH1998 <1335248067@qq.com>

huangtingwei9988 committed:
fix compressed_tensors bug

huangtingwei9988 committed:
revert code

huangtingwei9988 committed:
Use moe_wna16 kernel for compressed tensors wna16 moe models

huangtingwei9988 committed:
fix

huangtingwei9988 committed:
fix

huangtingwei9988 committed:
fix

yych0745 committed:
optimize kernel

yych0745 committed:
add moe w4a16 kernel

Commit:
wna16 marlin

huangtingwei9988 committed:
merge main

huangtingwei9988 committed:
merge main

huangtingwei9988 committed:
fix

huangtingwei9988 committed:
fix --quantization choices

AniZpZ committed:
update compressed tensor route

AniZpZ committed:
fix

AniZpZ committed:
fix

AniZpZ committed:
Merge remote-tracking branch 'origin/main' into w4a16_marlin

AniZpZ committed:
upd

AniZpZ committed:
upd

Commit:
support moe_marlin

AniZpZ committed:
Merge branch 'main' into w4a16_marlin

AniZpZ committed:
fix conflict

AniZpZ committed:
fix

AniZpZ committed:
fix

AniZpZ committed:
fix

AniZpZ committed:
test

AniZpZ committed:
format

merrymercy committed:
Merge branch 'main' into w4a16_marlin

huangtingwei9988 committed:
Merge branch 'main' into w4a16_marlin

AniZpZ committed:
Merge branch 'main' into w4a16_marlin

AniZpZ committed:
Merge remote-tracking branch 'origin/main' into w4a16_marlin

AniZpZ committed:
Merge remote-tracking branch 'origin/main' into w4a16_marlin

Commit:
PullRequest: 57 remove vllm dependency of wna16 marlin

Reviewed-by: 庄森 <zhuangsen.zp@antgroup.com>


* fix: remove part of vllm dependency and import error
* fix: modify the include package of gptq_marlin_repack
* fix: remove vllm dependency of fused_moe
* fix: delay the try_get_optimal_moe_config import
* fix: replace namespace of gptq_marlin_repack
* fix: update the head file of gptq_marlin_repack.cu
* update cmakelists
* modify the repack.cu
* modify namespace
* update head files
* update namespace marlin->marlin_moe_wna16
* update namespace
* add namespace
* update headfile
* remove invalid parentheses
* update CMakeLists.txt
* update headfiles
* update headsfile
* remove nested define
* add parentheses
* update
* move cuh to cu
* move identifier from namespace to outside
* update
* add namespace scope
* remove condition of compile define
* add null implementation for host
* remove namespace scope
* remove sm75
* remove define conditions & add gptq_marlin_repack_meta impl
* remove repack_meta
* add register namespace
* add namespace in sgl_kernel_ops
* add register namespace
* delay the moe_align_block_size import
* modify the import of moe_align_block_size
* add scalar_type.py & modify import of fused_moe.py
* add compilation condition & remove VLLM_AVAILABLE
* remove VLLM_AVAILABLE

AniZpZ committed:
upd

walker-ai committed:
remove vllm in moe_wna16

AniZpZ committed:
Merge remote-tracking branch 'origin/main' into w4a16_marlin

zhyncs committed:
Merge branch 'main' into w4a16_marlin

AniZpZ committed:
format

zhyncs commented:
@AniZpZ 
- please fix the conflicts
- may we submit the sgl-kernel related pr first

Commit:
PullRequest: 90 remove vllm dependency of awq_marlin

Merge branch 'wyt/fix_vllm_dependency of git@code.alipay.com:Theta/SGLang.git into w4a16_marlin

https://code.alipay.com/Theta/SGLang/pull_requests/90


Reviewed-by: 庄森 <zhuangsen.zp@antgroup.com>


* fix: remove vllm dependency in compressed_tensors_moe & add gptq_marlin_moe_repack in gptq
* update gptq_marlin_repack in gptq
* add condition of fp8_config in radixattention
* chore: remove vllm dependency & add kernel
* chore: remove vllm dependency & add kernel
* chore: remove vllm dependency & add kernel
* chore: remove gptq_marlin_gemm kernel
* add unit test
* add copyright & add unit test

Commit:
PullRequest: 91 fix PytestWarning & update quant_utils

Merge branch 'wyt/fix_vllm_dependency of git@code.alipay.com:Theta/SGLang.git into w4a16_marlin

https://code.alipay.com/Theta/SGLang/pull_requests/91

Reviewed-by: 庄森 <zhuangsen.zp@antgroup.com>

* fix: remove vllm dependency in compressed_tensors_moe & add gptq_marlin_moe_repack in gptq
* update gptq_marlin_repack in gptq
* add condition of fp8_config in radixattention
* chore: remove vllm dependency & add kernel
* chore: remove vllm dependency & add kernel
* chore: remove vllm dependency & add kernel
* chore: remove gptq_marlin_gemm kernel
* add unit test
* add copyright & add unit test
* chore: fix PytestWarning & update quant_utils

AniZpZ committed:
Merge remote-tracking branch 'origin/main' into w4a16_marlin

AniZpZ committed:
format

AniZpZ commented:
> @AniZpZ
> 
> * please fix the conflicts
> * may we submit the sgl-kernel related pr first

@zhyncs 
i submit a new pr only related with sgl-kernel https://github.com/sgl-project/sglang/pull/7683
